src/Chapter3.hs

```diff
@@ -1006,6 +1161,43 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+data FKActions =
+    FKAttack
+  | FKHealing
+  | FKDefense
+  deriving (Eq,Ord,Enum,Show)
+
+data FMActions =
+    FMAttack
+  | FMHealing
+  | FMDefense
+  deriving (Eq,Ord,Enum,Show)
+
+data FKnight = FKnight
+    {
+      fkAttack :: Int,
+      fkHealth :: Int,
+      fkDefense :: Int,
+      fkActions :: FKActions
+    }
+
+data FMonster = FMonster
+    {
+      fmAttack :: Int,
+      fmHealth :: Int,
+      fmActions :: FMActions
+    }
+
+class Fighter a where
+  attacked :: Int -> a -> a
+
+instance Fighter FKnight where
+  attacked :: Int -> FKnight -> FKnight
+  attacked mDmg k = k { fkHealth = fkHealth k + fkDefense k - mDmg }
+
+instance Fighter FMonster where
+  attacked :: Int -> FMonster -> FMonster
+  attacked kDmg m = m { fmHealth = fmHealth m - kDmg }
```

Comment:
Nice!
You can also write `fight` function to imitate the real battle ⚔️ 
For example:
```haskell
fight :: [FKActions] -> FKnight -> [FMActions] -> FMonster -> Winner
```